src/lua/init.lua

```diff
@@ -62,7 +62,9 @@ local function exit(code)
     -- Make sure we yield even if the code after
     -- os.exit() never yields. After on_shutdown
     -- fiber completes, we will never wake up again.
-    while true do fiber.yield() end
+    fiber.yield()
+    -- Don't eat up 100% CPU while on_shutdown triggers are running
+    while true do fiber.sleep(0.001) end
```

Comment:
Why do you need `fiber.yield`? Why do you use 1 ms timeout?

```Lua
local TIMEOUT_INFINITY = 500 * 365 * 86400
while true do fiber.sleep(TIMEOUT_INFINITY) end
```

Reply:
Indeed, it seems you may sleep indefinitely here

Reply:
Thanks, fixed.